src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/EntityAIRangeGuard.java

```diff
@@ -178,13 +188,48 @@ protected AIState searchTarget()
      */
     protected AIState huntDown()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "huntDown" is 17 which is greater than 15 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)